building-and-packaging/material/packaging_hpc_notes.md

```diff
@@ -0,0 +1,196 @@
+# Packaging for High-Performance Computing (Notes)
+
+## Spack Package Creation
+
+- Finish Spack setup
+
+  ```
+  spack compiler find
+  spack external find
+  ```
+
+  This will find preinstalled compilers and further software. This way we do not have to recompile everything from scratch.
+
+  **Note** `spack external find` is an experimental feature and might fail. System packages [can be defined manually](https://spack.readthedocs.io/en/latest/getting_started.html#system-packages).
+
+- Install a simple example package
+
+  ```bash
+  spack spec zlib
+  ```
+
+  This will trigger the bootstrap process of Spack. It installs `clingo` the dependency resolver.
+
+  ```bash
+  spack install --reuse zlib
+  ```
+
+  Will now install the package `zlib` by downloading and compiling it. `--reuse` tells it to reuse package that exist already if possible
+
+  ```bash
+  spack find
+  ```
+
+  will show `zlib` being available now and only `zlib` as the preinstalled software has not been used/added to the root environment of Spack yet.
+
+- Create base package (boilerplate)
+
+  ```bash
+  spack create https://github.com/Simulation-Software-Engineering/HelloWorld/archive/refs/tags/v0.3.0.tar.gz
+  ```
+
+  Spack will fetch the archive and try to deduce some basic settings. It will also check for other releases and asks whether to add checksums. We wann all checksums.
```

Comment:
```suggestion
  Spack will fetch the archive and try to deduce some basic settings. It will also check for other releases and asks whether to add checksums. We want all checksums.
```